plugins/volumeFilter_test.go

```diff
@@ -0,0 +1,439 @@
+package plugins
+
+import (
+	"database/sql"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/stellar/go/txnbuild"
+
+	hProtocol "github.com/stellar/go/protocols/horizon"
+	"github.com/stellar/kelp/model"
+	"github.com/stellar/kelp/support/postgresdb"
+	"github.com/stellar/kelp/support/utils"
+	"github.com/stretchr/testify/assert"
+)
+
+func connectTestDb() *sql.DB {
+	postgresDbConfig := &postgresdb.Config{
+		Host:      "localhost",
+		Port:      5432,
+		DbName:    "test_database",
+		User:      os.Getenv("POSTGRES_USER"),
+		SSLEnable: false,
+	}
+
+	_, e := postgresdb.CreateDatabaseIfNotExists(postgresDbConfig)
+	if e != nil {
+		panic(e)
+	}
+
+	db, e := sql.Open("postgres", postgresDbConfig.MakeConnectString())
+	if e != nil {
+		panic(e)
+	}
+	return db
+}
+
+var testNativeAsset hProtocol.Asset = hProtocol.Asset{Type: "native"}
+
+func TestMakeFilterVolume(t *testing.T) {
+	testAssetDisplayFn := model.AssetDisplayFn(func(asset model.Asset) (string, error) {
+		sdexAssetMap := map[model.Asset]hProtocol.Asset{
+			model.Asset("XLM"): testNativeAsset,
+		}
+		assetString, ok := sdexAssetMap[asset]
+		if !ok {
+			return "", fmt.Errorf("cannot recognize the asset %s", string(asset))
+		}
+		return utils.Asset2String(assetString), nil
+	})
+
+	testEmptyVolumeConfig := &VolumeFilterConfig{}
+	db := connectTestDb()
+
+	emptyConfigVolumeQuery, e := makeDailyVolumeByDateQuery(db, "", "native", "native", []string{}, []string{})
+	if e != nil {
+		t.Log("could not make empty config volume query")
+		return
+	}
+
+	marketIDs := []string{"marketID"}
+	marketConfigVolumeQuery, e := makeDailyVolumeByDateQuery(db, "", "native", "native", []string{}, marketIDs)
```

Comment:
Good call on the empty db. I'm not 100% certain that we can reconstruct the `sqlQuery` on `DailyVolumeByDate` without calling an extracted function like `makeDailyVolumeByDateQuery`, since the ultimate result looks like the below. Will come back to this.

`SELECT SUM(base_volume) as total_base_volume, SUM(counter_cost) as total_counter_volume FROM trades WHERE market_id IN ('4ba0ebdac9', 'marketID') AND account_id IN ('accountID') AND DATE(date_utc) = $1 and action = $2 group by DATE(date_utc)`